### ISSUE TITLE
Vespucci 15.0.5 has a matching problem with keys with subkeys or prefixes

### ISSUE THREAD

**Issue by jajajaneeneenee** [MarcusWolschon/osmeditor4android#1258]
## Vespucci Version
15.0.5

## Download source
Google Play Store (and GitHub, see below)

## Device (Manufacturer and Model)
Huawei Honor 7x

## Android Version 
9.1.0

## Behaviour/Symptoms
Since Vespucci 15.0.5, icons of objecs with keys with subkeys (prefixes, suffixes) like "disused:shop=yes" or "traffic_sign:forward=XYZ" are not displayed anymore and are shown as "unknown object (no preset found)" if editing. Matching doesn't seem to function. Normal objects with simple keys work as expected.

I deinstalled Vespucci v15.0.5 (from Google Play Store) and installed v15.0.4 from GitHub, and it worked with v15.0.4 (and older versions, I also tried v14.1.7 and v15.03). Then I installed 15.0.5 from GitHub again, and it didn't work.

See screenshots:

Vespucci 15.0.4:
![Screenshot_06_Vespucci_15_0_4](https://user-images.githubusercontent.com/48966999/96322895-34260380-101b-11eb-88ae-fff03fef71ae.jpg)

Vespucci 15.0.5:
![Screenshot_07_Vespucci_15_0_5](https://user-images.githubusercontent.com/48966999/96322910-3c7e3e80-101b-11eb-97bf-90b2e5dc6d13.jpg)

## Expected Behaviour
v15.0.5 should show the icons and match the objects like v15.0.4.

## How to recreate
First try with v15.0.4, then with v15.0.5.

## Crash dump submitted (no or yes + date)
no

## Any other potentially relevant information
no

**Comment by simonpoole** [comment]
This is a more or less a deliberate effect of https://github.com/MarcusWolschon/osmeditor4android/commit/917e48268789f34de25c41277084034642e5ebaf which is a fix for https://github.com/MarcusWolschon/osmeditor4android/issues/1241

However, this does have the effect that preset items that have a single tag without any predetermined values won't get matched at all (doesn't have anything to do with sub keys or not).

There's an alternative approach to suppressing the spurious matches that I'm looking at as a replacement for the above change, that should work with the key only matches too.

**Comment by simonpoole** [comment]
Regression fixed with different approach in https://github.com/MarcusWolschon/osmeditor4android/commit/233f3105aa583b4264c6dff96f310ee601fc6c1e

**Comment by jajajaneeneenee** [comment]
Thanks a lot for the fast fix! That's great!

Also for the explanation – because I also noticed just yesterday that normal keys (without subkeys/prefixes) without a fixed key/value scheme also don't match. I am looking forward for the next release!

**Comment by jajajaneeneenee** [comment]
> This is a more or less a deliberate effect of [917e482](https://github.com/MarcusWolschon/osmeditor4android/commit/917e48268789f34de25c41277084034642e5ebaf) which is a fix for #1241
> 
> However, this does have the effect that preset items that have a single tag without any predetermined values won't get matched at all (doesn't have anything to do with sub keys or not).

One addition: yesterday, I noticed, that "buildig:part=yes" does not match at all. But it HAS a predetermined value - not by a key-value element but by a combo element (in your "Beautified JOSM preset") – which works with other objects (if their key is in the object_keys).

So it worked after I added "building:part" to object_keys (but I am quite sure, that it worked in the past without being in the object_keys, but not 100% sure). And not sure if this is the same issue (or another side effect of [917e482](https://github.com/MarcusWolschon/osmeditor4android/commit/917e48268789f34de25c41277084034642e5ebaf)) and will be fixed with your fix. Or if this is a issue for the repository of the "Beautified JOSM preset". It's not always easy to decide this for a user like me (because it also happens with the build-in preset!). And also not always easy to decide when to open a new issue ...

And a note:
I also noticed, that items, which really only have a **text element** with a match attribute (which means no predetermined values at all), will never match. And that this is the ONLY item type which has a real matching problem. Adding to object_keys does not help here. On the other hand, items with key, combo, multiselect and check elements with a match attribute DO work (if they are added to object_keys). I strongly hope that this is a part of your fix here, because it would be very helpful, if such items would match (e.g. for traffic_sign:forward with many many possible values).

I will wait until 15.0.6 is released ... hope it comes soon ... then I can open a new issue if necessary. Only wanted to mention it here ... perhaps you will read it (although this issue is closed).
